stl/inc/type_traits

```diff
@@ -1473,6 +1473,14 @@ _NODISCARD constexpr conditional_t<!is_nothrow_move_constructible_v<_Ty> && is_c
     return _STD move(_Arg);
 }
 
+// Currently, invoke() isn’t depicted as constexpr in the Standard, and implementers are forbidden from “strengthening”
+// it to be constexpr, yet variant visit() is required to be constexpr and have invoke-like behavior. We solve this by
+// using a macro to stamp out a public non-constexpr and internal constexpr implementation.
+
+// Now that a Core Language issue affecting constexpr invoke() has been fixed (CWG1581), we may make this unconditional

```

Comment:
We wouldn't strengthen unconditionally, we would just make invoke call _C_invoke.

Reply:
I suppose, we'd also eliminate this macro and just expand it in the constexpr case right?

Reply:
There are throughput considerations here - `std::function` is throughput-sensitive and it uses `invoke()`. If `invoke()` were defined to call `_C_invoke()`, that would be an unnecessary layer of function calls.

I believe that the best thing to do would be to:

1. De-macroize this machinery,
2. Make the `_Invoker` layer unconditionally `constexpr`,
3. Make `invoke()` conditionally `_CONSTEXPR20`,
4. Change `apply()` and `visit()` to use the `_Invoker` layer.

Reply:
> Make the _Invoker layer unconditionally constexpr

Can we do that without CWG-1581?

Reply:
I don't think this conversation is relevant to documenting what's there _now_ right.

Reply:
This affects your comment "Now that a Core Language issue affecting constexpr std::invoke has been fixed (CWG-1581), we may make this unconditional and remove the macro (which violates the strengthening rule in C++14/17 mode). TRANSITION, P1065R2" which talks about the future.

Reply:
yeah it seems like we may be able to do this without CWG-1581...

Reply:
>yeah it seems like we may be able to do this without CWG-1581

I disagree; getting incorrect behavior from invoke is exactly why CWG-1581 existed in the first place. It's possible that Stephan's implementation, which does not use expression SFINAE, is immune to that problem, but I don't understand this area well enough to make such a firm statement. See http://www.open-std.org/JTC1/SC22/WG21/docs/papers/2018/p1065r0.html for more nitty gritty details.